### ISSUE TITLE
Contract.new promise is broken

### ISSUE THREAD

**Issue by kumavis** [ethjs/ethjs-contract#17]
resolves without result
```js
const txHash = await MainContract.new({ from: coinbase })
```

resolves without throwing
```js
const txHash = await MainContract.new()
```

works
```js
const txHash = await pify(MainContract.new)({ from: coinbase })
```

**Comment by SilentCicero** [comment]
Will review tomorrow.

**Comment by SilentCicero** [comment]
Should be addressed now @kumavis -- as of v1.9

**Comment by kumavis** [comment]
confirmed fixed
